taskmanager/cron_jobs.go

```diff
@@ -0,0 +1,58 @@
+package taskmanager
+
+import (
+	"context"
+	"time"
+)
+
+// CronJobHandler is the handler for a cron job
+type CronJobHandler func(ctx context.Context, target interface{}) error
+
+// CronJob definition, params reduced to the minimum, all required
+type CronJob struct {
+	Handler CronJobHandler
+	Period  time.Duration
+}
+
+// CronJobs as a map prevents duplicate jobs with the same name
+type CronJobs map[string]CronJob
+
+// CronJobsInit registers and runs the cron jobs
+func (tm *Client) CronJobsInit(target interface{}, cronJobsMap CronJobs) (err error) {
```

Comment:
does this `target` need to be `interface{}` (aka `any`)? can't it be replaced with some concrete type?

Reply:
A far as it is in separate package (`taskmanager`) - it is only a general utility. The main use case is to use it with `bux.client` which is in other package. So from `CronJobsInit` perspective it doesn't know the `bux.client` structure.